chef/cookbooks/crowbar-pacemaker/providers/order_only_existing.rb

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright 2015, SUSE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+action :create do
+  # evil command line; there must be a better way to fetch the list of resources
+  # unfortunately, "crm_resource --list-raw" doesn't list groups/clones
+  all_resources = %x{crm configure show | grep -vE '^\s|^node|id=' | cut -d ' ' -f 2}.split("\n")
```

Comment:
Well, there's `crm configure show xml` which is technically the best way to generate machine-readable output.  So you could do e.g.

    crm configure show xml | xpath '//primitive | //group | //clone'

However SP3's `xpath` is kind of lame and AFAIK can't extract attributes, so this approach isn't much more useful.

However I *don't* like your `grep -v` blacklist approach; I think it would be more reliable to `egrep '^(primitive|group|clone)'`.  Having said that, you can do all this with `awk`:

    crm configure show | awk '/^(primitive|group|clone)/ {print $2}'

so that would be my final suggestion.